### PR TITLE
go.mod: fix k8s.io/client-go version

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

/kind misc

Fix k8s.io/client-go version in go.mod, as the previous one targets a wrong version (retracted)

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
Fix k8s.io/client-go version in go.mod, as the previous one targets a wrong version (retracted)
```

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Does this actually change anything considering `k8s.io/client-go => k8s.io/client-go v0.22.5` ?

https://github.com/vdemeester/tektoncd-pipeline/blob/release-v0.34.x/go.mod#L8

File: go.mod [hunk 1]
```diff
@@ -30,7 +30,7 @@ require (
 	gomodules.xyz/jsonpatch/v2 v2.2.0
 	k8s.io/api v0.23.4
 	k8s.io/apimachinery v0.23.4
-	k8s.io/client-go v1.5.2
+	k8s.io/client-go v0.23.3
```
Comment (imjasonh) [inline go.mod:33]: The other deps are v0.23.4 -- should this match?

Comment (vdemeester) [reply go.mod:33]: oh yes my bad.. I can't type..

**Comment by 06kellyjac** [comment]
Why does the bot post multiple times? :upside_down_face:

**Comment by vdemeester** [comment]
> Why does the bot post multiple times? upside_down_face

That's a good question 😅

**Pull request merged**